### PR TITLE
Improving nnUNet inference speed

### PR DESCRIPTION
**Pull request by ancestor-mithril**
Main additions are:
* Using `torch.backends.cudnn.benchmark = True` (used during training, but forgot during inference).
* Replacing `torch.no_grad` context with a single `torch.inference_mode` context.
* Using inplace operations for calculating and using the gaussian.

### CODE REVIEW

**Review by FabianIsensee — CHANGES_REQUESTED** [review]

File: nnunetv2/inference/predict_from_raw_data.py [hunk 1]
```diff
@@ -574,24 +573,27 @@ def _internal_predict_sliding_window_return_logits(self,
             predicted_logits = torch.zeros((self.label_manager.num_segmentation_heads, *data.shape[1:]),
                                            dtype=torch.half,
                                            device=results_device)
-            n_predictions = torch.zeros(data.shape[1:], dtype=torch.half, device=results_device)
             if self.use_gaussian:
                 gaussian = compute_gaussian(tuple(self.configuration_manager.patch_size), sigma_scale=1. / 8,
                                             value_scaling_factor=10,
                                             device=results_device)
+                n_predictions = torch.zeros(data.shape[1:], dtype=torch.half, device=results_device)
```
[the remark this answers is not in the document]
Comment (ancestor-mithril) [reply nnunetv2/inference/predict_from_raw_data.py:580]: I see. Somehow, I failed to see the usage of `n_predictions` outside of the `self.use_gaussian` context. 
I'll fix this.

File: nnunetv2/inference/sliding_window_prediction.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from scipy.ndimage import gaussian_filter
 
 
-@lru_cache(maxsize=2)
+@lru_cache(maxsize=None)
```
[the remark this answers is not in the document]
Comment (ancestor-mithril) [reply nnunetv2/inference/sliding_window_prediction.py:10]: Ok, I got this. Normally, `compute_gaussian` should be called with the same parameters so only 1 or 2 values would be in the cache.

**Comment by FabianIsensee** [comment]
Thanks! Had to make two small changes to make sure everything works + yields the same results.
What would you say is the advantage of `@torch.inference_mode()` over `with torch.no_grad()`? Prediction speed seems to be the same

**Comment by ancestor-mithril** [comment]
Tensors created during inference mode are slimmer because they do not have a version counter and requires_grad can't be set to True anymore. More information here: https://stackoverflow.com/a/74197846/18441695.
In my tests,  `@torch.inference_mode()` is slightly less than 1% faster than `no_grad` for nnUNet.
